src/generic/infobar.cpp

```diff
@@ -84,11 +84,9 @@ bool wxInfoBarGeneric::Create(wxWindow *parent, wxWindowID winid)
     // button at the very right
     //
     // NB: AddButton() relies on the button being the last control in the sizer
-    //     and being preceded by a spacer
     wxSizer * const sizer = new wxBoxSizer(wxHORIZONTAL);
     sizer->Add(m_icon, wxSizerFlags().Centre().Border());
-    sizer->Add(m_text, wxSizerFlags().Centre());
-    sizer->AddStretchSpacer();
```

Comment:
I think `sizer->AddStretchSpacer();` should be left in because some user code (KiCad's) expects to find the spacer to remove buttons after it: https://gitlab.com/kicad/code/kicad/-/blob/8128f4ebd96b45ea517042c6b364e883de7e4ab4/common/widgets/wx_infobar.cpp#L320

Reply:
Ugh, KiCad really shouldn't have depended on this internal implementation detail.

@marekr Can you change this code or do we really have to preserve this sizer? I guess if we do, we'll have to document this as part of this class API.

Reply:
It would cause problems for people building older KiCad versions with latest wxWidgets, when bisecting for example.

Reply:
> It would cause problems for people building older KiCad versions with latest wxWidgets, when bisecting for example.

Meh? boost 1.86 already breaks bisects for us. Lol.

Reply:
Oh well, I've restored the spacer.

BTW, we definitely could add `RemoveAllButtons()` to `wxInfoBar` itself instead of reimplementing it in KiCad.

Reply:
IMO, if a way to remove all the buttons is added into the generic infobar in wx, then it is fine to remove the spacer for that version of wx - we can just conditionalize the code for this (we have done that a lot in the past anyway, so it isn't a big deal).